subprojects/tooling-api/tooling-api.gradle.kts

```diff
@@ -0,0 +1,121 @@
+    crossVersionTestRuntime(project(":buildComparison"))
+    crossVersionTestRuntime(project(":ivy"))
+    crossVersionTestRuntime(project(":maven"))
+}
+
+gradlebuildJava {
+    moduleType = ModuleType.ENTRY_POINT
+}
+
+testFixtures {
+    from(":core")
+    from(":logging")
+    from(":dependencyManagement")
+    from(":ide")
+}
+
+val jar: Jar by tasks
+
+val baseVersion: String by rootProject.extra
+
+val shadedJarWithoutVersion by tasks.creating(ShadedJar::class) {
+    val outputDir = file("$buildDir/shaded-jar-without-version")
+    sourceFiles = jar.outputs.files +
+        files(deferred { configurations.runtimeClasspath - configurations.publishCompile })
+    analysisFile = file("$outputDir/analysis.txt")
+    classesDir = file("$outputDir/classes")
+    jarFile = file("$outputDir/gradle-tooling-api-shaded-$baseVersion.jar")
+    keepPackages = setOf("org.gradle.tooling")
+    unshadedPackages = setOf("org.gradle", "org.slf4j", "sun.misc")
+    ignorePackages = setOf("org.gradle.tooling.provider.model")
+    shadowPackage = "org.gradle.internal.impldep"
+}
+
+val buildReceipt = tasks.getByPath(":createBuildReceipt")
+
+val toolingApiShadedJar by tasks.creating(Zip::class) {
+    destinationDir = file("$buildDir/shaded-jar")
+    dependsOn(shadedJarWithoutVersion, buildReceipt)
+    from(zipTree(shadedJarWithoutVersion.jarFile))
+    baseName = "gradle-tooling-api-shaded"
+    from(buildReceipt) {
+        into("/org/gradle")
+    }
+    extension = "jar"
+    version  = baseVersion
+}
+
+apply {
+    from("buildship.gradle")
+    from("eclipse.gradle")
+}
+
+val sourceJar: Jar by tasks
+
+sourceJar.run {
+    configurations.compile.allDependencies.withType<ProjectDependency>().forEach {
+        from(it.dependencyProject.java.sourceSets[SourceSet.MAIN_SOURCE_SET_NAME].groovy.srcDirs)
+        from(it.dependencyProject.java.sourceSets[SourceSet.MAIN_SOURCE_SET_NAME].java.srcDirs)
+    }
+}
+
+artifacts {
+    add("publishRuntime", mapOf(
+        "file" to toolingApiShadedJar.archivePath,
+        "name" to base.archivesBaseName,
+        "type" to "jar",
+        "builtBy" to toolingApiShadedJar
+    ))
+}
+
+tasks.create<Upload>("publishLocalArchives") {
+    val repoBaseDir = rootProject.file("build/repo")
+    configuration = configurations.publishRuntime
+    isUploadDescriptor = false
+    repositories {
+        ivy {
+            artifactPattern("$repoBaseDir/${project.group.toString().replace("\\.", "/")}/${base.archivesBaseName}/[revision]/[artifact]-[revision](-[classifier]).[ext]")
+        }
+    }
+
+    doFirst {
+        if (repoBaseDir.exists()) {
+            // Make sure tooling API artifacts do not pile up
+            repoBaseDir.deleteRecursively()
+        }
+    }
+}
+
+val integTestTasks: DomainObjectCollection<IntegrationTest> by extra
+
+integTestTasks.all {
+    binaryDistributions.binZipRequired = true
+    libsRepository.required = true
+}
+
+the<TestFileCleanUpExtension>().isErrorWhenNotEmpty = false
```

Comment:
:x: should use the generated static accessor:
```kotlin
testFilesCleanup {
    isErrorWhenNotEmpty = false
}
```

Reply:
Fixed.